Test views prototype

loysollivier opened the pull request:
First test views prototype, if you can have a look whenever you have the time. 
I tried to follow the new code framework as much as possible.

Thanks

loysollivier committed:
Implement the test suite view add test set data

Get and display the test set information (name, test-cases)

Signed-off-by: Loys Ollivier <lollivier@baylibre.com>

loysollivier committed:
Implement the test suite view, add test case data

Get and display the test case information (name, results)

Signed-off-by: Loys Ollivier <lollivier@baylibre.com>

loysollivier committed:
Create the links to the test set and case urls

Create the links to the future test set and test case urls
The links exists but they point to nothing as the urls are not
implemented yet

Signed-off-by: Loys Ollivier <lollivier@baylibre.com>

loysollivier committed:
New test view landing page

Created a new test view landing page. This page sorts the tests by name
and displays a count of test sets and cases available for each suite.
The code follows the same standard than other resources (boot, socs, ...)

Signed-off-by: Loys Ollivier <lollivier@baylibre.com>

loysollivier committed:
Test suite view update

Information on the test suite displayed in a new way

Signed-off-by: Loys Ollivier <lollivier@baylibre.com>

loysollivier committed:
Test landing page update

Sort the tests by available boards first

Signed-off-by: Loys Ollivier <lollivier@baylibre.com>

loysollivier committed:
New skeleton for test view sorted by tree

Create a new skeleton for the test view sorted by board and tree
Also update the file names to follow new naming standards

Signed-off-by: Loys Ollivier <lollivier@baylibre.com>

loysollivier committed:
New skeleton for the test report

Create a new skeleton for the test report
A test report is sorted by board, job, kernel

Signed-off-by: Loys Ollivier <lollivier@baylibre.com>

loysollivier committed:
New test view when sorted by board, job, kernel

New test view displaying the test results by lab for a specific
board, job and kernel

Signed-off-by: Loys Ollivier <lollivier@baylibre.com>

loysollivier committed:
Fix kernel field

Make sure the kernel field is valid

Signed-off-by: Loys Ollivier <lollivier@baylibre.com>

loysollivier committed:
Test view ordering

Enable paging and order the results latest first

Signed-off-by: Loys Ollivier <lollivier@baylibre.com>

loysollivier committed:
New test suite detailed view

For a specific test suite ID display the test suite details such as:
- Name, date, time,...
- test sets childs
- test cases childs

Signed-off-by: Loys Ollivier <lollivier@baylibre.com>

loysollivier committed:
Test board view, fix the tests results

Fixed test board view latest tests results.
Get the latest test suite for each tree/branch and
update the counts accordingly.

Signed-off-by: Loys Ollivier <lollivier@baylibre.com>

loysollivier committed:
Test board-job view, fix the tests results

Fixed test board-job view latest tests results.
Get the latest test suite for each tree/branch/commit and
update the counts accordingly.

Signed-off-by: Loys Ollivier <lollivier@baylibre.com>

loysollivier committed:
Test board-job-kernel view, fix the details

Update the details (board, tree, branch, git commit, ...) for the specific
board/job/kernel tests.

Signed-off-by: Loys Ollivier <lollivier@baylibre.com>

loysollivier committed:
Test suite-id view, fix the details

Update the details (status, errors, time, log, ...).
Show them as non available as they are not defined yet.

Signed-off-by: Loys Ollivier <lollivier@baylibre.com>

MiloCasagrande commented on app/dashboard/__init__.py:
```diff
@@ -315,6 +318,9 @@ def ajax_compare(doc_id, api):
     if validate_csrf(request.headers.get(CSRF_TOKEN_H, None)):
+        # Workaround because the backend uses /test/suite/...
+        if (resource == "suite"):
+            resource = "test/suite"
```
What would happen if you pass `test/suite` directly in the ajax call and not just `suite`? Did you try that?
I have no idea if that regex will catch it or if it will fail...

loysollivier replied:
Yes I did, it's not caught because the regex uses "/" as delimiters, sending one in the ajax call breaks the regex link and is not caught: E404. 
I could add an optional "/" in the regex for resource and it might work, I you prefer. It's a another solution.

MiloCasagrande replied:
Nah... that's fine, I was more wondering if it magically worked correctly. ;-)

MiloCasagrande commented on app/dashboard/static/js/app/view-socs-soc-job-kernel.2017.4.js:
```diff
@@ -672,7 +672,7 @@ define([
         gJob = document.getElementById('job-name').value;
     }
-    if (document.getElementById('job-name') !== null) {
+    if (document.getElementById('kernel-name') !== null) {
```
Ah! Good catch!

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-all.2017.7.2.js:
```diff
@@ -0,0 +1,395 @@
+/*!
+ * kernelci dashboard.
+ * 
+ * Copyright (C) 2014, 2015, 2016, 2017  Linaro Ltd.
```
You should add Baylibre up here as well.

loysollivier replied:
Added Baylibre and author

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-all.2017.7.2.js:
```diff
@@ -0,0 +1,395 @@
+        setTimeout(getTestsDone.bind(null, results), 25);
+        setTimeout(getSuitesCount.bind(null, results), 75);
+        setTimeout(getLabsCount.bind(null, results), 100);
+        setTimeout(enableSearch, 175);
```
Is there a reason for such differences in the timeout values? Are those operations so expensive?

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-board-job-kernel.2017.7.2.js:
```diff
@@ -0,0 +1,534 @@
+
+    function getTestsFail(lab) {
+        html.replaceContent(
+            document.getElementById("lab-" + lab),
```
Can you use single quotes here?

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-board-job-kernel.2017.7.2.js:
```diff
@@ -0,0 +1,534 @@
+        document.getElementById('li-test').setAttribute('class', 'active');
+    }, 15);
+
+   	gTableCount = {};
```
Indentation here seems a little bit off... is there a tab character?

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-board-job-kernel.2017.7.2.js:
```diff
@@ -0,0 +1,534 @@
+        var results;
+
+        function _createOp(result) {
+        	suiteId = result._id.$oid;
```
Same here for the indentation.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-board-job-kernel.2017.7.2.js:
```diff
@@ -0,0 +1,534 @@
+
+            $.when(deferred)
+                .fail(error.error, getBatchCountFail)
+                .done(function(response) {
```
You need to change this variable name, since `response` is already defined in the outer scope.
Even `resultz` would be fine, and pass it to the `getBatchCountDone` function call.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-board-job-kernel.2017.7.2.js:
```diff
@@ -0,0 +1,534 @@
+            $.when(deferred)
+                .fail(error.error, getBatchCountFail)
+                .done(function(response) {
+                	getBatchCountDone(labTable, response)
```
Missing a semicolon at the end here, plus weird indentation.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-board-job-kernel.2017.7.2.js:
```diff
@@ -0,0 +1,534 @@
+        function getData(lab) {
+            data.lab_name = lab;
+
+		    gLabTable.push(table({
```
Look for the indentation here as well.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-board-job-kernel.2017.7.2.js:
```diff
@@ -0,0 +1,534 @@
+		        tableLoadingDivId: 'lab-table-loading-' + lab,
+		        tableDivId: 'lab-table-div-' + lab
+		    }));
+		    var labTable = gLabTable[gLabTable.length - 1];
```
Better move the `var` declaration  outside of this function definition, where the other two variables are define, or at least at the beginning of this function.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-board-job-kernel.2017.7.2.js:
```diff
@@ -0,0 +1,534 @@
+        deferred = request.get('/_ajax/test/suite', data);
+        $.when(deferred)
+            // .fail(error.error, )
+            .done(updateDetails)
```
Missing a semicolon here at the end.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-board-job-kernel.2017.7.2.js:
```diff
@@ -0,0 +1,534 @@
+        // Update the general details
+        deferred = request.get('/_ajax/test/suite', data);
+        $.when(deferred)
+            // .fail(error.error, )
```
If you don't need the error handler, you can remove it.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-board-job-kernel.2017.7.2.js:
```diff
@@ -0,0 +1,534 @@
+        gFileServer = document.getElementById('file-server').value;
+    }
+
+    gLabList = document.getElementById('lab-names');
```
Hmmm... probably, in this case instead of going through a `split` call, it might be better to add multiple `input` elements to the HTML page, retrieve them via `querySelectorAll` and then apply a `map` function to extract the values (you actually need an extra step to convert what `querySelectorAll` returns, since it's not an Array). It would be something like:
```
Array.prototype.slice.call(document.querySelectorAll('input.lab-names')).map( ... );
```

In case this needs to be changed in the HTML template to add the multiple `input`s with a `class` name of `lab-names` (or something else).

MiloCasagrande commented on app/dashboard/templates/tests-board-job-kernel.html:
```diff
@@ -0,0 +1,129 @@
+    </div>
+</div>
+{%- if lab_names|length > 0 %}
+<input type="hidden" id="lab-names" value="{{ lab_names|sort|join(',') }}">
```
This is where the multiple `input`s should be done.
Inside the following for-loop, maybe added at the end of the for-loop.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-board-job.2017.7.2.js:
```diff
@@ -0,0 +1,411 @@
+            });
+        }
+
+        function _getTestSuiteDone(response) {
```
Better to change the variable name here, since `response` is already defined in the outside scope.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-board-job.2017.7.2.js:
```diff
@@ -0,0 +1,411 @@
+                sort: 'created_on',
+                sort_order: '-1',
+                limit: '1'
+            }
```
Missing a semicolon here at the end.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-board-job.2017.7.2.js:
```diff
@@ -0,0 +1,411 @@
+            deferred = request.get('/_ajax/test/suite', queryData);
+            $.when(deferred)
+                .done(_getTestSuiteDone)
+                .fail(function(response) {
```
Better to change variable name here as well.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-board-job.2017.7.2.js:
```diff
@@ -0,0 +1,411 @@
+            $.when(deferred)
+                .done(_getTestSuiteDone)
+                .fail(function(response) {
+                    ttest.getCountFail(result.vcs_commit)
```
Missing a semicolon here at the end.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-board.2017.7.2.js:
```diff
@@ -0,0 +1,457 @@
+        var batchOps;
+        var deferred;
+        var data;
+        var gBaseSelf = "board=" + gBoard;
```
Use single quotes here (and two lines later as well).

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-board.2017.7.2.js:
```diff
@@ -0,0 +1,457 @@
+        var data;
+        var gBaseSelf = "board=" + gBoard;
+        // TODO fix this
+        var gBaseOthers = "board=" + gBoard;
```
If not used, it can be removed.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-board.2017.7.2.js:
```diff
@@ -0,0 +1,457 @@
+        var deferred;
+        var suiteId;
+        var suiteTree;
+        var suiteBranch
```
Missing semicolon at the end.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-board.2017.7.2.js:
```diff
@@ -0,0 +1,457 @@
+            });
+        }
+
+        function _getTestSuiteDone(response) {
```
Better to change variable name here.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-board.2017.7.2.js:
```diff
@@ -0,0 +1,457 @@
+            deferred = request.get('/_ajax/test/suite', queryData);
+            $.when(deferred)
+                .done(_getTestSuiteDone)
+                .fail(function(response) {
```
Better to change variable name here.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-board.2017.7.2.js:
```diff
@@ -0,0 +1,457 @@
+                .fail(function(response) {
+                    document.getElementById('suites-count-'+ result.job)
+                        .innerHTML ='&infin;';
+                    ttest.getCountFail(result.job)
```
Missing semicolon here at the end.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-board.2017.7.2.js:
```diff
@@ -0,0 +1,457 @@
+    function getJobsDone(response) {
+        var columns,
+            results,
+            deferred;
```
This variable is not used, could be removed.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-board.2017.7.2.js:
```diff
@@ -0,0 +1,457 @@
+            {
+                aggregate: 'job',
+                // TODO add a date range
+                // date_range: gDateRange,
```
I guess this is needed or you won't get any data out now, but it should be set.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-suite-id.2017.7.2.js:
```diff
@@ -0,0 +1,557 @@
+/*!
+ * kernelci dashboard.
+ * 
+ * Copyright (C) 2014, 2015, 2016, 2017  Linaro Ltd.
```
You should add Baylibre here, or use the short version of the license header (although it might not be compliant, and we should have a COPYING file somewhere).

loysollivier replied:
Added Baylibre and author

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-suite-id.2017.7.2.js:
```diff
@@ -0,0 +1,557 @@
+    'utils/html',
+    'utils/const',
+    'utils/format',
+    'utils/date'
```
This is an old habit, but you can move the `utils/date` import (or any imports that does not return anything) into the `js/kci-*` file that `require`s this one.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-suite-id.2017.7.2.js:
```diff
@@ -0,0 +1,557 @@
+    'utils/const',
+    'utils/format',
+    'utils/date'
+], function($, init, error, request, u, bisect, ttest, html, appconst, format) {
```
The `format` import seems is not used.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-suite-id.2017.7.2.js:
```diff
@@ -0,0 +1,557 @@
+        // For each test case create a table row
+        // And update the matching set table
+        results.forEach(function(result) {
+            addCaseTableRow(result, docFrag)
```
Missing semicolon.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-suite-id.2017.7.2.js:
```diff
@@ -0,0 +1,557 @@
+            tableId = 'multiple-cases-table';
+            table = document.getElementById(tableId);
+            setTimeout(
+                _tableMessage.bind(
```
Where is this function defined?

loysollivier replied:
Missed the hunk during the add, it's there now: https://github.com/loysollivier/kernelci-frontend/blob/fa78179f546b2ba864b3695d73426264cdcab211/app/dashboard/static/js/app/view-tests-suite-id.2017.7.2.js#L46

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-suite-id.2017.7.2.js:
```diff
@@ -0,0 +1,557 @@
+        var results;
+        var table;
+        var tableId;
+        var tableBody;
```
Variable is never used.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-suite-id.2017.7.2.js:
```diff
@@ -0,0 +1,557 @@
+
+        tableNode = document.createElement('table');
+        tableNode.className =
+            'table table-striped table-condensed \
```
Need to close the string and concatenate with a `+`, or:
```
tableNode.className = 'table table-striped table-condensed';
tableNode.className += ' table-hover clickable-table hidden';
```

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-suite-id.2017.7.2.js:
```diff
@@ -0,0 +1,557 @@
+             table-hover clickable-table hidden'
+        tableNode.id = 'table-set-' + setId;
+        tableCaption = tableNode.createCaption();
+        tableCaption.innerHTML = setName; 
```
There is a white space at the end of the line.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-suite-id.2017.7.2.js:
```diff
@@ -0,0 +1,557 @@
+            html.errorDiv('Error loading data.'));
+    }
+
+    function getTestSetAndCaseData(response) {
```
You can remove `response` here if it's not needed.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-suite-id.2017.7.2.js:
```diff
@@ -0,0 +1,557 @@
+        var resultSetLength;
+        var deferredCase;
+        var resultCaseLength;
+        var result;
```
Variable never used.

MiloCasagrande reviewed (changes requested):
Wow, big work here! There are a few JavaScript fixes that needs to be applied, but overall the work is really good!

loysollivier committed:
Code clean up after review

Code clean up, there's still a change to do on the lab enumeration.
Will be part of a separate commit.
https://github.com/kernelci/kernelci-frontend/pull/14#pullrequestreview-58598687

Signed-off-by: Loys Ollivier <lollivier@baylibre.com>

loysollivier committed:
Test view by board-job-kernel code review

Parse the list of labs using a querySelectorAll.

Signed-off-by: Loys Ollivier <lollivier@baylibre.com>

loysollivier commented:
Done reviewing the changes https://github.com/kernelci/kernelci-frontend/pull/14#pullrequestreview-58598687

loysollivier committed:
Test suite view by ID add log link

Add links to the log for the test suite view by ID.
This log is unique and regroups the boot + test. A possible improvement
would be to parse it and add on overlay for test case recognition.

Signed-off-by: Loys Ollivier <lollivier@baylibre.com>

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-board-job-kernel.2017.7.2.js:
```diff
@@ -518,12 +519,14 @@ require([
+    Array.prototype.slice.call(document.querySelectorAll('input.lab-names'))
+        .map(extendLabList);
+    if (gLabList) {
         if (gLabList.length > 0) {
```
Minor tweak.

You can remove the previous check on `gLabList` and only check on its length. In Javascript an empty array will still evaluate to true, but since you are already setting it up as an array at the beginning of the file, you can just check for the length.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-suite-id.2017.7.2.js:
```diff
@@ -0,0 +1,631 @@
+        result = response.result[0];
+        bootLog = null;
+
+        function bootLogDone(bootLog, bootResponse) {
```
You are passing `bootLog` that is defined in the outer scope: you can avoid passing it and just access it from inside the function scope.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-suite-id.2017.7.2.js:
```diff
@@ -0,0 +1,631 @@
+            deferred = request.get('/_ajax/boot', {id: result.boot_id.$oid});
+            $.when(deferred)
+                .done(function(bootResponse) {
+                    bootLogDone(bootLog, bootResponse);
```
Removing the `bootLog` argument from the `bootLogDone` function, this `.done` call can be simplified into:
```
.done(bootLogDone);
```

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-suite-id.2017.7.2.js:
```diff
@@ -0,0 +1,631 @@
+        var job;
+        var kernel;
+        var lab;
+        var logsNode;
```
This variable is not used.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-suite-id.2017.7.2.js:
```diff
@@ -0,0 +1,631 @@
+        var result;
+        var smallNode;
+        var spanNode;
+        var statusNode;
```
This variable is not used.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-suite-id.2017.7.2.js:
```diff
@@ -0,0 +1,631 @@
+        var tableHead;
+        var tableBody;
+        var tableRow;
+        var tableCell;
```
This variable is not used.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-suite-id.2017.7.2.js:
```diff
@@ -0,0 +1,631 @@
+        var cellNode;
+        var rowNode;
+        var caseName;
+        var logsNode;
```
This variable is not used.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-board-job-kernel.2017.7.2.js:
```diff
@@ -0,0 +1,537 @@
+    }
+
+    function extendLabList(lab) {
+        gLabList.push(lab.value)
```
Missing the semicolon at the end.

MiloCasagrande commented on app/dashboard/static/js/app/view-tests-board-job-kernel.2017.7.2.js:
```diff
@@ -0,0 +1,537 @@
+    Array.prototype.slice.call(document.querySelectorAll('input.lab-names'))
+        .map(extendLabList);
+    if (gLabList) {
+        if (gLabList.length > 0) {
```
You can squeeze this check into just one line, and only check for `gLabList` length.
An empty array in Javascript evaluates to true.

MiloCasagrande reviewed (changes requested):
A few minor changes, but it's already looking great!
Good work!

loysollivier committed:
Change review and synchronous calls

Made the changes from: https://github.com/kernelci/kernelci-frontend/pull/14#discussion_r136522997
Made the calls to display the log in cascade so they are not impacted by
the ajax ansynchronous results.

Signed-off-by: Loys Ollivier <lollivier@baylibre.com>

loysollivier commented:
Requested changes are done
